### ISSUE TITLE
Panache support for not initilized proxies

### ISSUE THREAD

**Issue by suchwerk** [quarkusio/quarkus#9602]
**Description**
The use-case is to add not-initilized proxies to a set inside a panache entity. Right now i have to inject the entity manager to access getReference. It would be nice if PanacheEntityBase itself would implement getReference.

So 

```
    @Inject
    EntityManager entityManager;
    ids.forEach(id->list.add(entityManager.getReference(Item.class, id)) );
``` 

to be replaced by something like:

```   
    ids.forEach(id->list.add(Item.getReference(id)));
```

**Comment by Sanne** [comment]
I agree, this would be very useful. Thanks for opening!

@FroMage : I suppose this is doable?

**Comment by FroMage** [comment]
You can use `Panache.getEntityManager()` as well. But yeah, probably we could add this feature, but is it really common enough, do you think @Sanne ?

**Comment by suchwerk** [comment]
``` 
 ids.forEach(id->list.add(Item.getReference(id)));
``` 
looks leaner vs.
``` 
 ids.forEach(id->list.add(Panache.getEntityManager().getReference(Item.class, id)));
```

**Comment by FroMage** [comment]
Sure, I don't question that. But we don't want to add every Hibernate feature in a Panache layer because that would make the API huge and scary. So we always try to resist additions to the most common use-cases that people need. Hence my question to @Sanne but perhaps @emmanuelbernard has an idea too.

**Comment by Sanne** [comment]
You can't easily update collections without this .. so yes I'd say it's commonly useful. As many other things :)

**Comment by Sanne** [comment]
Although TBH `Panache.getEntityManager().getReference(..` doesn't look too bad - your call @FroMage

**Comment by FroMage** [comment]
If I give you `Entity getReference(Object id)`, I'll have to give you `List<Entity> getReferences(Object... ids)` too because ` ids.forEach(id->list.add(Item.getReference(id)));` is just yuck compared to `Item.getReferences(ids)`.

**Comment by emmanuelbernard** [comment]
Hum, `getReference` is a relatively useful optimisation but using it to get loads of entities to add to a `List` is less common.
Personally I'm getting scared of Panache's own weight and would resist adding these two features. But I might in my ivory tower.

**Comment by loicmathieu** [comment]
I tend to agree with @emmanuelbernard, then people will want `findReference(query, params)` with all the variations because, you usually list via a query not via known ids ...

@suchwerk what do you do with those references you get ? Maybe we're missing a different functionality here ?

**Comment by suchwerk** [comment]
@loicmathieu I just want to add them to a jointable.

**Comment by ryckowski** [comment]
I'm using myEntity.setRelated(getReference(Related.clas, id)) quite often to set the related entity and there's no reason fetch it. From sql point of view just want to set foreign key id which I know
Is there a better practise?

**Comment by gavinking** [comment]
> Hum, `getReference` is a relatively useful optimisation but using it to get loads of entities to add to a `List` is less common.

I agree but it looks like the issue title and the mentioned usage is a bit misleading?

All that's described is adding `Item.getReference(id)`, which is definitely needed IMO. (Nothing specific to lists in that.)

**Comment by gaetanBloch** [comment]
Just discussed this issue with @yrodiere and will work on a PR on Panache in aner future
